fixed bug "all ip port is not available" #3700

soulmz opened the pull request:
## Purpose of the pull request

close #3700 
<!--(For example: This pull request adds spotless plugin).-->

## Brief change log

<!--*(for example:)*
  - *Add spotless-maven-plugin to root pom.xml*
-->
- fixed bug "all ip port is not available" , 获取 k8s service list 列表失败
- modify  kubernetes 拼写错误。

## Verify this pull request

<!--*(Please pick either of the following options)*-->

- 因环境因素，本地和线上代码已经替换了 gateway jar 包，运行正常。

<img width="1594" alt="image" src="https://github.com/user-attachments/assets/d0b8eddd-2a3d-43df-8710-a16422879e9a">

<!--*(example:)*
  - *Added dinky-core tests for end-to-end.*
  - *Added UDFUtilTest to verify the change.*
  - *Manually verified the change by testing locally.* -->

Commit:
add feature table mapping-routes 映射路由
add feature table 正则表达式替换 表名匹配

Commit:
add feature table mapping-routes 映射路由
add feature table 正则表达式替换 表名匹配

Commit:
add doc 文档字段解释。

Commit:
rm 无用 flink cdc test

Commit:
rm 无用 flink cdc test

Commit:
add cdcsource  sink.table.mapping-routes 、sink.table.replace.pattern` 和 `sink.table.replace.with 文档案例

Commit:
add cdcsource  sink.table.mapping-routes 、sink.table.replace.pattern` 和 `sink.table.replace.with 文档案例

Commit:
add cdcsource  sink.table.mapping-routes 、sink.table.replace.pattern` 和 `sink.table.replace.with 测试 mapping-routes 和 pattern、with 优先级用例边界。

Commit:
add cdcsource  sink.table.mapping-routes 、sink.table.replace.pattern` 和 `sink.table.replace.with 测试 mapping-routes 和 pattern、with 优先级用例边界。

Commit:
fixed unit test
modify 注释中文改英文

Commit:
fixed unit test
modify 注释中文改英文

Commit:
modify pattern

Commit:
modify pattern

Commit:
modify pattern

Commit:
modify pattern

Commit:
modify pattern

Commit:
modify doc 文档

Commit:
modify doc 文档

Commit:
modify doc 文档

Commit:
modify doc 文档 正则的规则匹配方式

Commit:
modify doc 文档 正则的规则匹配方式

Commit:
checkstyle format

Commit:
checkstyle format

Commit:
test code format

Commit:
test code format

Commit:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
fix(kubernetes): 更正Kubernetes应用操作网关类的拼写错误

重命名KubetnetsApplicationOperatorGateway为KubernetesApplicationOperatorGateway以修正拼写错误，并更新相关引用。
此外，修复了潜在的bug，该bug可能导致无法找到服务列表，方法是在检索服务时明确指定命名空间。
还对日志消息进行了修改，以提高清晰度并反映部署成功时的jobName和clusterId。
在KubernetesOperatorGateway中，还修正了变量名和日志引用的拼写错误。

Commit:
fix(kubernetes): 修正服务列表日志级别并处理空指针异常

- 将服务列表的打印日志级别从info更改为debug，以避免过多的日志输出影响性能。
- 在查询服务列表之前添加了空指针检查，以防止在列表为空时引发异常。

此更改确保在处理Kubernetes服务列表时的稳定性，并优化了日志记录，使其仅在调试时输出服务列表信息。

soulmz committed:
Merge branch 'DataLinkDC:dev' into dev

Commit:
fix(kubernetes): 修复无法查找服务列表的问题

解决了无法查找服务列表的错误，当尝试在KubernetesApplicationOperatorGateway中使用带有指定服务名称时。该修复确保了在指定名称空间中正确检索服务列表。

更改日志:
调整了kubernetesClient.services()的调用，增加了对问题PR #3700的引用，以提供对修复上下文的更好理解。

Commit:
fix(kubernetes): 修复无法查找服务列表的问题
servicePort.getName.equals -> endsWith
解决了无法查找服务列表的错误，当尝试在KubernetesApplicationOperatorGateway中使用带有指定服务名称时。该修复确保了在指定名称空间中正确检索服务列表。
更改日志:
调整了kubernetesClient.services()的调用，增加了对问题PR #3700的引用，以提供对修复上下文的更好理解。

Commit:
fix(kubernetes): 修复无法查找服务列表的问题

解决了无法查找服务列表的错误，当尝试在KubernetesApplicationOperatorGateway中使用带有指定服务名称时。该修复确保了在指定名称空间中正确检索服务列表。

更改日志:
调整了kubernetesClient.services()的调用，增加了对问题PR #3700的引用，以提供对修复上下文的更好理解。

Commit:
Merge remote-tracking branch 'origin/dev' into dev

gaoyan1998 commented on build-extr.sh:
```diff
@@ -56,7 +56,7 @@ while getopts "j:f:" opt; do
 
 # 构建命令
-CMD="./mvnw clean package -Dmaven.test.skip=true -P prod,${JDK_VERSION},flink-single-version,aliyun,${FLINK_VERSION},web"
+CMD="./mvnw clean package -Dmaven.test.skip=true -P prod,${JDK_VERSION},flink-single-version,scala-2.12,aliyun,${FLINK_VERSION},web"
```
dev has deleted the scala version selection. The default is 2.12. There is no need to add it here.

Commit:
fix(kubernetes): 解决服务列表为空的问题

- 通过增加时间单位等待时长 2 分钟，针对大作业时间不足够。延长 5 分钟。()
- KubernetesApplicationOperatorGateway 中找不到服务列表的问题并且服务列表为空，则抛出异常提示检查svc列表是否存在。

Commit:
fix(build-extr): 修正JDK版本格式并执行构建命令- 将JDK_VERSION变量的值从"jdk11"更改为"jdk-11"，以符合正确的格式。
-确保构建命令CMD中正确使用了JDK_VERSION变量。
-移除执行构建命令时 scala profile

此更改解决了上一个构建中与JDK版本指定相关的问题，并确保构建命令按预期执行。

gaoyan1998 reviewed (approved):
LGTM
